var/spack/repos/builtin/packages/aomp/package.py

```diff
@@ -4,104 +4,121 @@
+    "014fca1fba54997c6db0e84822df274fb6807698b6856da4f737f38f10ab0e5d",
+    "ee146cff4b9ee7aae90d7bb1d6b4957839232be0e7dab1865e0ae39832f8f795"
+]
+
+# Used only for 3.5.0
+hip = [
+    "86eb7749ff6f6c5f6851cd6c528504d42f9286967324a50dd0dd54a6a74cacc7"
+]
+
+vdi = [
+    "b21866c7c23dc536356db139b88b6beb3c97f58658836974a7fc167feb31ad7f"
+]
+
+opencl = [
+    "8963fcd5a167583b3db8b94363778d4df4593bfce8141e1d3c32a59fb64a0cf6"
+]
+
+versions = ['3.5.0', '3.7.0', '3.8.0', '3.9.0', '3.10.0']
+versions_dict = dict()
+hashes = [aomp, devlib, llvm, flang, extras]
+hashes_35 = [aomp, devlib, llvm, flang, extras, hip, vdi, opencl]
+components = ['aomp', 'devlib', 'llvm', 'flang', 'extras']
+components_35 = [
+    'aomp', 'devlib', 'llvm', 'flang', 'extras', 'hip', 'vdi', 'opencl'
+]
+
+for outer_index, item in enumerate(versions):
+    if item == '3.5.0':
+        use_components = components_35
+        use_hashes = hashes_35
+    else:
+        use_components = components
+        use_hashes = hashes
+    for inner_index, component in enumerate(use_hashes):
+        versions_dict.setdefault(item, {})[use_components[inner_index]] = \
+            use_hashes[inner_index][outer_index]
 
 
 class Aomp(Package):
     """llvm openmp compiler from AMD."""
 
     homepage = tools_url + "/aomp"
-    url = tools_url + "/aomp/archive/rocm-3.9.0.tar.gz"
+    url = tools_url + "/aomp/archive/rocm-3.10.0.tar.gz"
 
     maintainers = ['srekolam', 'arjun-raj-kuppala', 'estewart08']
-    version('3.9.0', sha256=aomp38['aomp'])
-    version('3.8.0', sha256=aomp38['aomp'])
-    version('3.7.0', sha256=aomp37['aomp'])
-    version('3.5.0', sha256=aomp35['aomp'])
+    version('3.10.0', sha256=versions_dict['3.10.0']['aomp'])
+    version('3.9.0', sha256=versions_dict['3.9.0']['aomp'])
+    version('3.8.0', sha256=versions_dict['3.8.0']['aomp'])
+    version('3.7.0', sha256=versions_dict['3.7.0']['aomp'])
+    version('3.5.0', sha256=versions_dict['3.5.0']['aomp'])
+
+    # Cmake above 3.18 would fail the build on 3.5.0
+    depends_on('cmake@3:', type='build')
+    depends_on('cmake@3:3.17', when='@3.5.0', type='build')
+
+    # Python 2 is needed for 3.5.0 and 3.8.0, limit py-setuptools
+    # to avoid spec error
+    depends_on('python@2.7:2.8', when='@3.5.0:3.8.0', type='build')
+    depends_on('py-setuptools@44', when='@3.5.0:3.8.0',
+               type='build')
+
+    depends_on('python@3:', type='build', when='@3.9.0:')
+    depends_on('py-setuptools', when='@3.9.0:', type='build')
 
-    depends_on('cmake@3.5.2:3.13.4', type='build')
     depends_on('mesa18~llvm@18.3:', type=('build', 'link'))
-    depends_on('py-setuptools@44.1.0', type='build')
-    depends_on('python@2.7.18', type='build', when='@3.5.0:3.8.0')
-    depends_on('python@3.6.9', type='build', when='@3.9.0:')
     depends_on('py-pip', when='@3.8.0:', type='build')
-    depends_on('py-wheel@0.29.0', when='@3.8.0:', type=('build', 'run'))
+    depends_on('py-wheel', when='@3.8.0:', type=('build', 'run'))
```

Comment:
Not sure if I've asked this in the past, but are pip and wheel actually needed?

Reply:
Yes, they are needed during the ompd portion of the openmp build.